receiver/pulsarreceiver/config.go

```diff
@@ -20,7 +20,7 @@ type Config struct {
 	// The Subscription that receiver will be consuming messages from (default "otlp_subscription")
 	Subscription string `mapstructure:"subscription"`
 	// Encoding of the messages (default "otlp_proto")
-	Encoding string `mapstructure:"encoding"`
+	EncodingID *component.ID `mapstructure:"encoding"`
```

Comment:
Let's keep the option name clear. We also need to keep the old config option as deprecated for now to avoid introducing a significant breaking change. I believe you can still remove the code from this component but keep this option
```suggestion
	EncodingExtension *component.ID `mapstructure:"encoding_extension"`
```